packages/builders/src/components/Components.ts

```diff
@@ -1,29 +1,33 @@
-import { APIMessageComponent, ComponentType } from 'discord-api-types/v9';
-import { ActionRow, ButtonComponent, Component, SelectMenuComponent } from '../index';
+import { APIMessageComponent, APIModalComponent, ComponentType } from 'discord-api-types/v9';
+import { ActionRow, ButtonComponent, Component, SelectMenuComponent, TextInputComponent } from '../index';
 import type { MessageComponent } from './ActionRow';
 
 export interface MappedComponentTypes {
 	[ComponentType.ActionRow]: ActionRow;
 	[ComponentType.Button]: ButtonComponent;
 	[ComponentType.SelectMenu]: SelectMenuComponent;
+	[ComponentType.TextInput]: TextInputComponent;
 }
 
 /**
  * Factory for creating components from API data
  * @param data The api data to transform to a component class
  */
 export function createComponent<T extends keyof MappedComponentTypes>(
-	data: APIMessageComponent & { type: T },
+	data: (APIMessageComponent | APIModalComponent) & { type: T },
 ): MappedComponentTypes[T];
-export function createComponent<C extends MessageComponent>(data: C): C;
-export function createComponent(data: APIMessageComponent | MessageComponent): Component {
+export function createComponent<C extends MessageComponent | APIModalComponent>(data: C): C;
+export function createComponent(data: APIMessageComponent | APIModalComponent): Component {
 	switch (data.type) {
 		case ComponentType.ActionRow:
-			return (data instanceof ActionRow ? data : new ActionRow(data)) as Component;
+			// @ts-expect-error
+			return data instanceof ActionRow ? data : new ActionRow(data);
 		case ComponentType.Button:
-			return (data instanceof ButtonComponent ? data : new ButtonComponent(data)) as Component;
+			return data instanceof ButtonComponent ? data : new ButtonComponent(data);
 		case ComponentType.SelectMenu:
-			return (data instanceof SelectMenuComponent ? data : new SelectMenuComponent(data)) as Component;
+			return data instanceof SelectMenuComponent ? data : new SelectMenuComponent(data);
+		case ComponentType.TextInput:
+			return new TextInputComponent(data);
```

Comment:
Why doesn't this check if data is an instance of TextInputComponent like all the other cases do?